### PR TITLE
Added a market event to do non-item related stuff

### PR DESCRIPTION
**Pull request by linaori**
Example usage:
```lua 
{
    price = 1000, 
    name = 'buff-biters', 
    description = 'Makes enemy biters stronger\n+10% attack\n+10% armor', 
    name_label = 'Buff enemy Biters', 
    type = 'buff', 
    sprite = 'entity/medium-biter'
}
Event.add(Retailer.events.on_market_purchase, function (event)
    if event.item.type ~= 'buff' then
        return
    end
    event.player.print(format('Purchased %dx %s', event.count, event.item.name_label))
end)
```
image

image

### CODE REVIEW

File: features/retailer.lua [hunk 1]
```diff
@@ -18,10 +18,22 @@ local market_frame_close_button_name = Gui.uid_name()
 local item_button_name = Gui.uid_name()
 local count_slider_name = Gui.uid_name()
 local count_text_name = Gui.uid_name()
+local raise_event = script.raise_event
```
Comment (mheguy) [inline features/retailer.lua]: Can I bug you to define `raise_event` with the other function definitions just a few lines up?

**Comment by linaori** [comment]
All fixed. One thing I was curious about, perhaps granting the item should also go to an event listener, but not sure. It's now an if within the button pressed

**Comment by linaori** [comment]
Made the text align vertically centered
image

**Comment by Valansch** [comment]
Looks good :+1:

**Pull request merged**